applications/GeoMechanicsApplication/tests/test_line_interface_elements.py

```diff
@@ -81,6 +82,71 @@ def assert_outputs_for_3_plus_3_line_interface_element(self, simulation):
             self.assertAlmostEqual(relative_displacement_vertical_element[index][1], expected_vertical_displacement)
 
 
+    def assertVectorsAlmostEqual(self, actual_vectors, expected_vectors):
+        for actual_vector, expected_vector in zip(actual_vectors, expected_vectors):
+            self.assertVectorAlmostEqual(actual_vector, expected_vector)
+
+
+    def test_multi_stage_3_plus_3_line_interface_element_with_dirichlet_conditions(self):
+        file_path = test_helper.get_file_path(os.path.join('line_interface_elements', 'Dirichlet_multi_stage'))
+
+        initial_cwd = os.getcwd()
+        os.chdir(file_path)
+
+        project_parameters_file_names = ['ProjectParameters_stage1.json', 'ProjectParameters_stage2.json']
+        prescribed_displacement_vectors = [[-8.8933333333333332e-5, -2.22e-5, 0.0], [-2.0e-4, -4.44e-5, 0.0]]
+        for file_name, displacement_vector in zip(project_parameters_file_names, prescribed_displacement_vectors):
+            stage = test_helper.make_geomechanics_analysis(self.model, os.path.join(file_path, file_name))
+
+            stage.Run()
```

Comment:
Just for my understanding. Is the reason we can't use test_helper.run_stages because we need to do the asserts for each stage before running the next?

Reply:
Yes, that is exactly why I have set it up this way. The point is that we need to check results by looking at the database, since not all output items can be written to result files yet. When it its possible to write all needed output items to result files, we can simplify the test code further.